### PR TITLE
Removing the need to manually declare symbols, and wrap them in unsafe.

### PR DESCRIPTION
**Pull request by DavidDeSimone**
This works similar to how our setup for emacs_globals works.

It would be nice if we also used this to generate LispObject equivalents for these symbols, i.e. `LispObject(Qnil)` so we don't have to litter the codebase with LispObject::from_raw(Qhash_table_p) etc, but we can tackle that in the future.

### CODE REVIEW

**Comment by DavidDeSimone** [comment]
Actually I have a different idea for this. I will close this and update later.

File: rust_src/remacs-sys/build.rs [hunk 1]
```diff
@@ -103,33 +110,60 @@ fn generate_globals() {
...
     let out_path = PathBuf::from(env::var("OUT_DIR").unwrap()).join("globals.rs");
     let mut out_file = File::create(out_path).expect("Failed to create definition file");
-
+    let mut parse_state = ParseState::ReadingGlobals;
+    
     write!(out_file, "#[allow(unused)]\n").expect("Write error!");
     write!(out_file, "#[repr(C)]\n").expect("Write error!");
     write!(out_file, "pub struct emacs_globals {{\n").expect("Write error!");
 
+
```
Comment (shaleh) [inline rust_src/remacs-sys/build.rs]: dead line

**Comment by shaleh** [comment]
Looking good

**Comment by brotzeit** [comment]
This looks really nice. I would like to bring up [this topic](https://github.com/Wilfred/remacs/issues/345) again, while we're at it. Do we really need a huge list of lisp functions in lib.rs ? I can live with keeping it, but I think we should consider this. We can add them to the corresponding files after all lisp functions are ported, but I think this is useless at this point. And the merge conflicts are really annoying...

**Comment by shaleh** [comment]
Agreed. What are the steps to resolve it? Want to open an issue to track? Or can you just submit a PR and be done? Because I am +10 on this

On October 26, 2017 12:12:50 PM PDT, brotzeit <notifications@github.com> wrote:
>This looks really nice. I would like to bring up [this
>topic](https://github.com/Wilfred/remacs/issues/345) again, while we're
>at it. Do we really need a huge list of lisp functions in lib.rs ? I
>can live with keeping it, but I think we should consider this. We can
>add them to the corresponding files after all lisp functions are
>ported, but I think this is useless at this point. And the merge
>conflicts are really annoying...
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/Wilfred/remacs/pull/415#issuecomment-339769786

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by brotzeit** [comment]
I don't know how to do it, I just assumed it's possible ;)
But sure, I'll give it a try.

**Comment by shaleh** [comment]
PR #429 shows how to move `defsubr`.

**Comment by shaleh** [comment]
Trigger build

**Pull request merged**